### PR TITLE
Update the spec: include iframe header and relaxed filtering

### PR DESCRIPTION
**Pull request by xyaoinum**
Update the spec to include <iframe> request header and relaxed filtering behavior. Those new behaviors were added after the spec was drafted. https://github.com/patcg-individual-drafts/topics/pull/145 https://github.com/patcg-individual-drafts/topics/pull/143

### CODE REVIEW

**Comment by xyaoinum** [comment]
@domenic PTAL. Thanks!

File: spec.bs [hunk 1]
```diff
@@ -402,14 +408,28 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
...
+  Topics can be sent in the HTTP header for {{WindowOrWorkerGlobalScope/fetch()}} requests and for <a href="https://html.spec.whatwg.org/multipage/iframe-embed-object.html#navigate-an-iframe-or-frame">iframe navigation</a> requests. The response header for a topics related request can specify whether the caller should to be recorded.
 
   <h3 id="send-browsing-topics-header-boolean-associated-with-request-header">send browsing topics header boolean associated with Request</h3>
   A [=request=] has an associated <dfn for=request>send browsing topics header boolean</dfn>. Unless stated otherwise it is unset.
 
   <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
 
+  <h3 id="browsing-topics-content-attribute-for-iframe-element-header">browsingtopics content attribute for HTMLIframeElement</h3>
+  The <a href="https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element">iframe</a> element contains a <dfn for="content attribute">browsingtopics</dfn> <a spec=html>content attribute</a>. The IDL attribute `browsingTopics` <a spec=html>reflects</a> the [=content attribute/browsingtopics=] <a spec=html>content
```
Comment (domenic) [inline spec.bs:420]:
```suggestion
  The <a href="https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element">iframe</a> element contains a <dfn element-attr for="iframe">browsingtopics</dfn> <a spec=html>content attribute</a>. The IDL attribute <dfn attribute for="HTMLIFrameElement">browsingTopics</dfn> <a spec=html>reflects</a> the <{iframe/browsingtopics}> <a spec=html>content 
```

File: spec.bs [hunk 2]
```diff
@@ -402,14 +408,28 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
...
+  The <a href="https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element">iframe</a> element contains a <dfn for="content attribute">browsingtopics</dfn> <a spec=html>content attribute</a>. The IDL attribute `browsingTopics` <a spec=html>reflects</a> the [=content attribute/browsingtopics=] <a spec=html>content 
+  attribute</a>.
+
+  <pre class=idl>
+  interface HTMLIFrameElement : HTMLElement {
+    // existing attributes ...
+
+    [CEReactions] attribute boolean browsingTopics;
+  };
+  </pre>
```
Comment (domenic) [inline spec.bs:429]:
```suggestion
  partial interface HTMLIFrameElement {
    [CEReactions] attribute boolean browsingTopics;
  };
  </pre>
```

File: spec.bs [hunk 3]
```diff
@@ -430,6 +450,13 @@ spec: html; urlPrefix: https://www.rfc-editor.org/rfc/
 
   <span class=XXX>TODO: make the modification directly to the fetch spec.</span>
 
+  <h3 id="modification-to-create-navigation-params-by-fetching-steps-header">Modification to "create navigation params by fetching" steps</h3>
+  The following step will be added to the <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#create-navigation-params-by-fetching">create navigation params by fetching steps</a>, after step "Let |request| be a new [=Request/request=], with ...":
+
+  1. If <var ignore=''>navigable</var>'s [=container=] is an <a href="https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element">iframe</a> element, and if it has <a spec=html>content attribute</a> [=content attribute/browsingtopics=], then set |request|'s [=request/send browsing topics header boolean=] to true.
```
Comment (domenic) [inline spec.bs:456]:
```suggestion
  1. If <var ignore=''>navigable</var>'s [=container=] is an <{iframe}> element, and if it has a <{iframe/browsingtopics}> <a spec=html>content attribute</a>, then set |request|'s [=request/send browsing topics header boolean=] to true.
```

**Review by domenic — APPROVED** [review]
Looks good with editorial issues around defining and linking content and IDL attributes. I think my suggestions will suffice to fix them, but you might need a bit of tweaking... Relevant docs at https://speced.github.io/bikeshed/#dfn-types and https://speced.github.io/bikeshed/#autolink-wrappers .

Feel free to re-request review or ping if necessary.

**Pull request merged**